staging/src/kubevirt.io/client-go/api/v1/types.go

```diff
@@ -355,7 +355,7 @@ type VirtualMachineInstanceMigrationState struct {
 	// The address of the target node to use for the migration
 	TargetNodeAddress string `json:"targetNodeAddress,omitempty"`
 	// The list of ports opened for live migration on the destination node
-	TargetDirectMigrationNodePorts map[int]int `json:"targetDirectMigrationNodePorts,omitempty"`
+	TargetDirectMigrationNodePorts map[string]int `json:"targetDirectMigrationNodePorts,omitempty"`
```

Comment:
This is an api breaking change, which drastically complicates the situation for us. If we can't handle this in a backwards compatible way somehow, this little change will require about 10x the work. we'll have to increment our api version, and create a conversion webhook. Then we'll have to deal with the fact that we still need to support the old version somehow.

Reply:
actually. I believe I'm wrong here. This map is treated the same whether the key is an int or a string when we json marshal/unmarshal it.

Here's a simple program I wrote to prove it works. 

```
package main
        
import (
        "encoding/json"                                                                     
        "fmt"
        "strings"
)       
        
type a struct {
        SomeMap map[int]int `json:"someMap,omitempty"`
}                                                                                            
                                                                                       
type b struct {
        SomeMap map[string]int `json:"someMap,omitempty"`
}

func main() {

        mapAsInt := a{
                SomeMap: make(map[int]int),
        }
        
        mapAsString := b{
                SomeMap: make(map[string]int),
        }
        
        mapAsInt.SomeMap[123] = 456                                                                                            
        mapAsString.SomeMap["123"] = 456
        
        bytesMapAsInt, err := json.Marshal(mapAsInt)
        if err != nil {
                panic(err)
        }
        
        bytesMapAsString, err := json.Marshal(mapAsString)
        if err != nil { 
                panic(err)
        }
        
        fmt.Println(string(bytesMapAsInt))                                                                                     
        fmt.Println(string(bytesMapAsString))
                                                                                                                               
        if strings.Compare(string(bytesMapAsInt), string(bytesMapAsString)) == 0 {                                             
                fmt.Printf("equal\n")
        } else {
                fmt.Printf("not equal\n")                                                                                      
        }                                                                                                                      
}

```